ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/monitor-sessions.j2

```diff
@@ -0,0 +1,71 @@
+{# Leaf connected_endpoint monitor-sessions #}
+{% if monitor_sessions_data | length > 0 %}
+monitor_sessions:
+{%     for sname, group in monitor_sessions_data | groupby("name") %}
+{%         set group_roles = group | map(attribute='role', default='notdefined') | list %}
+{%         if 'source' in group_roles and 'destination' in group_roles %}
+  - name: {{ sname }}
+    sources:
+{%             for session in group | selectattr('role', 'equalto', 'source') %}
+{%                 if session.interface is arista.avd.defined %}
+      - name: {{ session.interface }}
+{%                     if session.source_settings is arista.avd.defined %}
+{%                         if session.source_settings.direction is arista.avd.defined %}
+        direction: {{ session.source_settings.direction }}
+{%                         endif %}
+{%                         if session.source_settings.access_group is arista.avd.defined %}
+        access_group:
+{%                             if session.source_settings.access_group.type is arista.avd.defined %}
+          type: {{ session.source_settings.access_group.type }}
+{%                             endif %}
+{%                             if session.source_settings.access_group.name is arista.avd.defined %}
+          name: {{ session.source_settings.access_group.name }}
+{%                             endif %}
+{%                             if session.source_settings.access_group.priority is arista.avd.defined %}
+          priority: {{ session.source_settings.access_group.priority }}
+{%                             endif %}
+{%                         endif %}
+{%                     endif %}
+{%                 endif %}
+{%             endfor %}
+    destinations:
+{%             set ns = namespace() %}
+{%             for session in group | selectattr('role', 'equalto', 'destination') %}
+{%                 if session.interface is arista.avd.defined %}
+      - {{ session.interface }}
+{%                 endif %}
+{%                 set ns.merged_settings = {} | combine(ns.merged_settings | arista.avd.default({}),
```

Comment:
session_settings should also be taken from source entries if it is set. So I suggest that you merge the settings outside of the filtered loops.
You can just put the list on the combine filter and skip namespace etc.
If you place this after line 4 in this template, you will save a lot of logic.
```jinja
{% set merged_settings = {} | combine(group, recursive=true, list_merge='append')
```

Reply:
I think you pointed this out from the beginning, but I did not even consider that somebody might want to do that (take general settings from source)..sorry!